src/main/java/io/gravitee/resource/cache/redis/RedisDelegate.java

```diff
@@ -75,12 +87,26 @@ public Object value() {
     public void put(Element element) {
         logger.debug("Put in cache {}", element.key());
         try {
-            cache.put(element.key(), element.value());
+            int ttl = this.timeToLiveSeconds;
+            if ((ttl == 0 && element.timeToLive() > 0) || (ttl > 0 && element.timeToLive() > 0 && ttl > element.timeToLive())) {
+                ttl = element.timeToLive();
+            }
+            RedisCacheWriter redisCacheWriter = (RedisCacheWriter) this.getNativeCache();
+
+            redisCacheWriter.put(getName(), buildKey(element.key()), this.serializer.serialize(element.value()), Duration.ofSeconds(ttl));
         } catch (Throwable e) {
             logger.error("Cannot put element in cache", e);
         }
     }
 
+    private byte[] buildKey(Object key) {
+        String allKey = this.getName() + key.toString().replaceAll("_", ":");
```

Comment:
Maybe we could add a `String keySeparator()` method on `Cache` interface (default to `_`) so we could adapt the `CachePolicy` and avoid the `replaceAll` invocation and any side effects (ex: if my custom key contains an `_`, it will be replaced as well).